test/001_highsnr_stdio_audio/test_rx.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Dec 23 07:04:24 2020
+
+@author: DJ2LS
+"""
+
+import ctypes
+from ctypes import *
+import pathlib
+import pyaudio
+import audioop
+import sys
+import logging
+import time
+import threading
+import sys
+import argparse
+
+#--------------------------------------------GET PARAMETER INPUTS  
+parser = argparse.ArgumentParser(description='Simons TEST TNC')
+parser.add_argument('--bursts', dest="N_BURSTS", default=0, type=int)
+parser.add_argument('--framesperburst', dest="N_FRAMES_PER_BURST", default=0, type=int)
+parser.add_argument('--mode', dest="FREEDV_MODE", default=14, type=int)
+parser.add_argument('--audioinput', dest="AUDIO_INPUT", default=0, type=int)  
+parser.add_argument('--debug', dest="DEBUGGING_MODE", action="store_true")  
+
+args = parser.parse_args()
+
+N_BURSTS = args.N_BURSTS
+N_FRAMES_PER_BURST = args.N_FRAMES_PER_BURST
+AUDIO_INPUT_DEVICE = args.AUDIO_INPUT
+MODE = args.FREEDV_MODE
+DEBUGGING_MODE = args.DEBUGGING_MODE
+
+
+# AUDIO PARAMETERS
+AUDIO_FRAMES_PER_BUFFER = 2048 
+MODEM_SAMPLE_RATE = 8000
+AUDIO_SAMPLE_RATE_TX = 48000
+
+# check if we want to use an audio device then do an pyaudio init
+if AUDIO_INPUT_DEVICE != 0: 
+    p = pyaudio.PyAudio()
+    stream_rx = p.open(format=pyaudio.paInt16, 
+                            channels=1,
+                            rate=AUDIO_SAMPLE_RATE_RX,
+                            frames_per_buffer=AUDIO_FRAMES_PER_BUFFER,
+                            input=True,
+                            input_device_index=AUDIO_INPUT_DEVICE,
+                            ) 
+
+# LOAD FREEDV
+libname = "libcodec2.so"
+c_lib = ctypes.CDLL(libname)
+
+# ctypes function init        
+
+c_lib.freedv_open.argype = [c_int]
+c_lib.freedv_open.restype = c_void_p
+
+c_lib.freedv_get_bits_per_modem_frame.argtype = [c_void_p]
+c_lib.freedv_get_bits_per_modem_frame.restype = c_int
+
+c_lib.freedv_nin.argtype = [c_void_p]
+c_lib.freedv_nin.restype = c_int
+
+c_lib.freedv_rawdatarx.argtype = [c_void_p, c_char_p, c_char_p]
+c_lib.freedv_rawdatarx.restype = c_int
+
+c_lib.freedv_get_n_max_modem_samples.argtype = [c_void_p]
+c_lib.freedv_get_n_max_modem_samples.restype = c_int
+
+c_lib.freedv_set_frames_per_burst.argtype = [c_void_p, c_int]
+c_lib.freedv_set_frames_per_burst.restype = c_void_p
+      
+c_lib.freedv_get_rx_status.argtype = [c_void_p]
+c_lib.freedv_get_rx_status.restype = c_int      
+
```

Comment:
@DJ2LS these are likely to be common to many programs. Could we put these "function headers" (using a C term) into a common file/module that we can import into many programs?

Reply:
seems to be solved with latest commit

Reply:
functions can be accessed by `codec2.api.freedv_open(12)` for example.